File: frontend/src/app/services/feedback.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Feedback } from '../models/feedback.model';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { User } from '../models/user.model';

@Injectable({
  providedIn: 'root'
})
export class FeedbackService {

  private apiUrl = `${environment.apiUrl}/feedback`;

  constructor(private http: HttpClient) { }

  createFeedback(userId:Number,feedback: Feedback): Observable<Feedback> {
    return this.http.post<Feedback>(this.apiUrl+"/user/"+userId, feedback);
  }

  getFeedbackByUserId(userId: number): Observable<Feedback[]> {
    return this.http.get<Feedback[]>(`${this.apiUrl}/user/${userId}`);
  }

  getFeedbackById(feedbackId: number): Observable<Feedback> {
    return this.http.get<Feedback>(`${this.apiUrl}/${feedbackId}`);
  }

  deleteFeedback(feedbackId: number): Observable<Feedback> {
    return this.http.delete<Feedback>(`${this.apiUrl}/${feedbackId}`);
  }

  getAllFeedback(): Observable<Feedback[]> {
    return this.http.get<Feedback[]>(this.apiUrl);
  }
  getUserById(userId: number): Observable<User>{
    return this.http.get<User>(this.apiUrl+"/"+userId) ///newMethod
  }

 
}

File: springapp/src/main/java/com/examly/springapp/service/FeedbackService.java
package com.examly.springapp.service;

import java.util.List;

import com.examly.springapp.model.Feedback;
import com.examly.springapp.model.User;

public interface FeedbackService {

public Feedback creatFeedback(Long userId, Feedback feedback);
public Feedback getFeedBackById(Long feedbackId);
public List<Feedback> getAllFeedback();
public Feedback deleteFeedback(long feedbackId);
public List<Feedback> getFeedbackByUserId(long userId);
public User getUserById(Long userId);


    
}
